### ISSUE TITLE
"Lost XXX Edge" error when generating visualisation

### ISSUE THREAD

**Issue by karl** [sverweij/state-machine-cat#15]
First up, thank you so much for State Machine Cat. It is awesome 🎉

I was able to add auto generated state machine visualisations to our code base in just a few hours thanks to your great work!

### Expected Behavior

When using the script:

```
APP,
PLAYER {
  RESTART_STREAM {
    PAUSED => PLAYING: play;
    PLAYING => PAUSED : pause;
  };

  LIVE_STREAM => PLAYING : switch to restart;
  RESTART_STREAM => LIVE_STREAM: stream finished;
  RESTART_STREAM => CONFIRM_SWITCH_TO_LIVE: start to live;
  CONFIRM_SWITCH_TO_LIVE => PLAYING: cancel switch to live;
  CONFIRM_SWITCH_TO_LIVE => LIVE_STREAM: confirm switch to live;
};

APP => LIVE_STREAM: start playback;
PLAYER => APP: on close;
```

I would expect it to generate a visualisation.

### Current Behavior

The following error occurs:

```
Error: lost LIVE_STREAM PLAYING edge
```

When testing using https://state-machine-cat.js.org/ with the default settings.

### Steps to Reproduce (for bugs)

Copy the script above into https://state-machine-cat.js.org/

### Context

We have started using state machines to model our more complex async UI state management. We use unit tests and State Machine Cat to auto generate up to date visualisations for our state machines.

I've been investigating adding support for nested states to our code and wanted to see what our state machine would look like with them. While converting the script to use nested states I spotted the error above.

### Your Environment

Using https://state-machine-cat.js.org/ on Chrome on macOS. I also get the same bug when running State Machine Cat from node.

**Comment by sverweij** [comment]
@karl thanks for raising this issue!

I've reproduced it. state-machine-cat uses [graphviz](http://www.graphviz.org) - which is also the origin of this error message. When I feed the graphviz ('dot') program directly to the graphviz command line, it gives the same error - but supplanted with a hint to use (graphviz' dialect of) html instead of graphviz' native records for nodes.

I've tried to follow the hint hacking on the generated dot and it seems to resolve the issue, but I'll need some time to (1) make it work in the reporter and (2) find a solution that plays nice with other constructs (entries, exits)

Sample output after my local `dot` hack 
<img width="696" alt="with-html-things" src="https://user-images.githubusercontent.com/4822597/39141123-11c1a46c-4727-11e8-9fce-775650f48aed.png">

(in case you wonder I've added the `entry` trigger manually to get a feel for how it could look)

**Comment by sverweij** [comment]
@karl I've created a first version of a fix which is being published as a beta to npm as I write this. You can install it with `npm i state-machine-cat@2.6.5-beta-0`. Let me know if it works for you!

It needs some polishing, but at least your script generates a correct visualisation again. I will update the online interpreter (and the atom plugin) after that polishing is done.

<img width="735" alt="screen shot 2018-04-23 at 23 13 23" src="https://user-images.githubusercontent.com/4822597/39153651-04a80b84-474c-11e8-8704-ef7685a3add1.png">

**Comment by karl** [comment]
Wow, thanks for the fast response on this!

I’ll take a look at the beta tomorrow and see how it works with our full state machine.

**Comment by karl** [comment]
All looks good with the beta, I'm able to render our full state machine locally.

**Comment by sverweij** [comment]
@karl Great, thanks for testing and getting back that fast! 

In the mean time I've been tweaking a bit so diagrams look (often indistinguishably) close to what they looked in version 2.6.4 and before (e.g. arrows touch the nodes, states are similar sized again if possible, ...). I'll try & publish to npm as a regular release (2.6.5) tomorrow - and update the online interpreter in the process.

B.t.w. what you describe in 'context' looks intriguing. If you have more to share - consider me interested :-)

**Comment by sverweij** [comment]
Ah, so for generating documentation. What's the reason to do that in the unit test(s)?

(I'm just interested in the concepts. I don't expect you to open source proprietary code :-) ).

In other news: Yesterday I've published state-machine-cat version 2.6.5 to npm & updated the online interpreter - with which this issue is

**Comment by karl** [comment]
I put the visualisation generation in the unit tests because it was quick and meant that other devs didn’t need to do anything new to update the docs.

But I’m not sure it is the right place to put it. I’m thinking of trying out a pre commit hook, or a separate command. And then running the same step on CI failing the build if the
